core/dispatch.c

```diff
@@ -1803,20 +1806,23 @@ adjust_syscall_continuation(dcontext_t *dcontext)
                 dcontext->asynch_target);
         }
 #    endif
-    } else if (vsyscall_syscall_end_pc != NULL &&
-               /* PR 341469: 32-bit apps (LOL64) on AMD hardware have
-                * OP_syscall in a vsyscall page
-                */
-               get_syscall_method() != SYSCALL_METHOD_SYSCALL) {
+    } else if (vsyscall_syscall_end_pc != NULL) {
+        /* PR 341469: 32-bit apps (LOL64) on AMD hardware have
+         * OP_syscall in a vsyscall page
```

Comment:
But this comment went with the exclusion for the method being SYSCALL -- so it doesn't make sense stranded here now?

Reply:
I think we should remove this comment. On AMD vsyscall page has OP_syscall and we should take care of it.
May be comparison was wrong and we should update `dcontext->asynch_target` when **syscall_method_is_syscall** (on all hardwares)?

Reply:
OK, so you are removing this comment, but please add a new one before line 1818 below to explain the AMD check there.